frontend/src/logic/schedule-logic/metadata.logic.ts

```diff
@@ -1,15 +1,19 @@
 import { VerboseDate, WeekDay } from "../../common-models/month-info.model";
-import { MetadataProvider } from "../schedule-provider";
+import { Sections } from "../providers/schedule-provider.model";
 import { DataRow } from "./data-row";
 import { MonthInfoLogic } from "./month-info.logic";
 import { BaseSectionLogic } from "./section-logic.model";
 import { MetaDataSectionKey } from "../section.model";
+import { MetadataProvider } from "../providers/metadata-provider.model";
 export enum MonthLogicActionType {
   UpdateFrozenDates = "updateFrozenDates",
 }
 
 export class MetadataLogic extends BaseSectionLogic implements MetadataProvider {
-  sectionKey = MetadataLogic.name;
+  get sectionKey(): keyof Sections {
+    return "Metadata";
```

Comment:
Wiem, że wygłąda jak zahardkodowany string, ale tak nie jest, ponieważ `sectionKey` jest zdefiniowany jako `keyof Sections` co zabrania podanie w tym miejscu klucza który do tych nie należy do tego interfejsu.

Reply:
Spoko, ale moim zdaniem jeśli używasz stringu "Metadata" w wielu miejscach, a tak jest pomimo refaktoru, to powinno być to wyłączone do jakiejś stałej

Reply:
To jest w pewien sposób zapisane w stałej. Tą stałą jest nazwa klucza w interfejsie Sections. Kompilator sprawdza czy to co jest wpisane w tym "zhardkowdowanym" stringu zgadza się z kluczami z interfejsu. Zrób sobie test i dodaj jakiś znak do tego stringu - to nie przejdzie kompilacji, więc IMO jest git.